storm-core/src/jvm/backtype/storm/topology/base/BaseWindowedBolt.java

```diff
@@ -0,0 +1,184 @@
+
+    /**
+     * Tuple count based sliding window configuration.
+     *
+     * @param windowLength    the number of tuples in the window
+     * @param slidingInterval the number of tuples after which the window slides
+     */
+    public BaseWindowedBolt withWindow(Count windowLength, Count slidingInterval) {
+        return withWindowLength(windowLength).withSlidingInterval(slidingInterval);
+    }
+
+    /**
+     * Tuple count and time duration based sliding window configuration.
+     *
+     * @param windowLength    the number of tuples in the window
+     * @param slidingInterval the time duration after which the window slides
+     */
+    public BaseWindowedBolt withWindow(Count windowLength, Duration slidingInterval) {
+        return withWindowLength(windowLength).withSlidingInterval(slidingInterval);
+    }
+
+    /**
+     * Time duration and count based sliding window configuration.
+     *
+     * @param windowLength    the time duration of the window
+     * @param slidingInterval the number of tuples after which the window slides
+     */
+    public BaseWindowedBolt withWindow(Duration windowLength, Count slidingInterval) {
+        return withWindowLength(windowLength).withSlidingInterval(slidingInterval);
+    }
+
+    /**
+     * Time duration based sliding window configuration.
+     *
+     * @param windowLength    the time duration of the window
+     * @param slidingInterval the time duration after which the window slides
+     */
+    public BaseWindowedBolt withWindow(Duration windowLength, Duration slidingInterval) {
+        return withWindowLength(windowLength).withSlidingInterval(slidingInterval);
+    }
+
+    /**
+     * A tuple count based window that slides with every incoming tuple.
+     *
+     * @param windowLength the number of tuples in the window
+     */
+    public BaseWindowedBolt withWindow(Count windowLength) {
+        return withWindowLength(windowLength).withSlidingInterval(new Count(1));
+    }
+
+
+    /**
+     * A time duration based window that slides with every incoming tuple.
+     *
+     * @param windowLength the time duration of the window
+     */
+    public BaseWindowedBolt withWindow(Duration windowLength) {
+        return withWindowLength(windowLength).withSlidingInterval(new Count(1));
+    }
+
+    /**
+     * A count based tumbling window.
+     *
+     * @param count the number of tuples after which the window tumbles
+     */
+    public BaseWindowedBolt withTumblingWindow(Count count) {
+        return withWindowLength(count).withSlidingInterval(count);
+    }
+
+    /**
+     * A time duration based tumbling window.
+     *
+     * @param duration the time duration after which the window tumbles
+     */
+    public BaseWindowedBolt withTumblingWindow(Duration duration) {
+        return withWindowLength(duration).withSlidingInterval(duration);
+    }
+
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+    }
```

Comment:
Can you put in a comment in all the empty methods, something like ```\\NOOP``` just so it is obvious that it is intended to be left blank.